app/uk/gov/hmrc/helplinefrontend/controllers/SelectNationalInsuranceServiceController.scala

```diff
@@ -42,8 +42,12 @@ class SelectNationalInsuranceServiceController @Inject()(implicit
     val result = SelectNationalInsuranceServiceForm.apply().bindFromRequest.fold(
       errors => BadRequest(selectNationalInsuranceService(errors)),
       {
-        case FindNiNumber => NotImplemented
-        case OtherQueries => NotImplemented
+        case FindNiNumber => request.session.get("HELPLINE_ORIGIN_SERVICE") match {
+          case Some("IV") => Redirect(s"${appConfig.findYourNationalInsuranceNumberFrontendUrl}/find-your-national-insurance-number/checkDetails?origin=IV")
+          case Some("PDV") => Redirect(s"${appConfig.findYourNationalInsuranceNumberFrontendUrl}/find-your-national-insurance-number/checkDetails?origin=PDV")
+          case None | Some(_) => Redirect(s"${appConfig.findYourNationalInsuranceNumberFrontendUrl}/find-your-national-insurance-number/checkDetails")
+        }
+        case OtherQueries => Redirect("/helpline/NATIONAL-INSURANCE")
```

Comment:
can this be a reverse route rather than hard coded, easier to amend in the future and easier to read to know where its going
i.e.` Redirect(routes.CallHelpdeskController.getHelpdeskPage(key here)`
and you might find you can pass in the NationalInsurance case object from models. which should automatically upper case and hyphenate (with scala magic) (see object CallOption)